cbirrt2/test/test_params.cpp

```diff
@@ -0,0 +1,123 @@
+/* File: test_params.cpp
+ * Author: Chris Dellin <cdellin@gmail.com>
+ * Copyright: 2015 Carnegie Mellon University
+ * License: BSD
+ */
+
+#include <openrave/openrave.h>
+#include <openrave-core.h>
+
+#include <gtest/gtest.h>
+
+#include "../stdafx.h"
+
+namespace {
+// from Johannes Schaub - litb
+// http://bloglitb.blogspot.de/2011/12/access-to-private-members-safer.html
+template<typename Tag, typename Tag::type M> struct Rob { friend typename Tag::type get(Tag) { return M; } };
+#define ROBBER(c,t,m) \
+   struct c ## _ ## m { typedef t c::*type; friend t c::*get(c ## _ ## m); }; \
+   template struct Rob<c ## _ ## m, &c::m>; \
+   t & rob_ ## m(c & i) { return i.*get(c ## _ ## m()); } \
+   const t & rob_ ## m(const c & i) { return i.*get(c ## _ ## m()); }
+ROBBER(TaskSpaceRegionChain, bool, bSampleGoalFromChain)
+ROBBER(TaskSpaceRegionChain, bool, bSampleStartFromChain)
+ROBBER(TaskSpaceRegionChain, bool, bConstrainToChain)
+ROBBER(TaskSpaceRegionChain, std::string, mimicbodyname)
+ROBBER(TaskSpaceRegionChain, RobotBasePtr, _pmimicbody)
+ROBBER(TaskSpaceRegionChain, std::vector<int>, _mimicinds)
+} // anonymous namespace
```

Comment:
Haha yes, although I feel like accessing private members for unit tests is one of the least objectionable uses for this d-: